src/uwu.py

```diff
@@ -158,8 +168,27 @@ def on_compile_and_run(self, code_editor: CodeEditor):
             self.code_panel.update_error_logs(errors=[])
             code_editor.start(editor=code_editor, compiler_status=compiler_status, update_logs_callback=self.code_panel.update_compiler_logs)
 
+    def update_analyzer_tabs_and_logs(self, code_editor: CodeEditor, lx_res, p_res, a_res):
+        if lx_res and p_res and a_res:
+            self.analyzer_panel.update_lexer(tokens=code_editor.tokens)
+            if code_editor.program:
+                self.analyzer_panel.update_parser_tree(program=code_editor.program)
+
+        self.code_panel.update_compiler_logs(editor=code_editor, is_compiling=False)
+
+        if len(code_editor.lx_errors) > 0:
+            self.analyzer_panel.clear_parser_tree()
+            self.code_panel.update_error_logs(errors=code_editor.lx_errors)
+        elif len(code_editor.p_errors) > 0:
+            self.code_panel.update_error_logs(errors=code_editor.p_errors)
+        elif len(code_editor.a_errors) > 0:
+            self.code_panel.update_error_logs(errors=code_editor.a_errors)
+        else:
+            self.code_panel.update_error_logs(errors=[])
+
```

Comment:
may not be necessary, `on_compiler_run` can have an extra parameter to use quick mode para hindi `code_editor.start()` ang gamitin niya